cmd/ci-operator/main.go

```diff
@@ -1197,12 +1206,31 @@ func summarizeRef(refs prowapi.Refs) string {
 	return fmt.Sprintf("Resolved source https://github.com/%s/%s to %s@%s", refs.Org, refs.Repo, refs.BaseRef, shorten(refs.BaseSHA, 8))
 }
 
-func eventRecorder(kubeClient *coreclientset.CoreV1Client, namespace string) record.EventRecorder {
+func eventRecorder(kubeClient *coreclientset.CoreV1Client, authClient *authclientset.AuthorizationV1Client, namespace string, dry bool) (record.EventRecorder, error) {
+	if dry {
+		return &record.FakeRecorder{}, nil
+	}
+	res, err := authClient.SelfSubjectAccessReviews().Create(&authapi.SelfSubjectAccessReview{
+		Spec: authapi.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authapi.ResourceAttributes{
+				Namespace: namespace,
+				Verb:      "create",
+				Resource:  "events",
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not check permission to create events: %v", err)
+	}
+	if !res.Status.Allowed {
+		log.Println("warning: Events will not be created because of lack of permission")
```

Comment:
Nit: looks like all other output lines start with a capital letter.

Reply:
`/shrug`
```
$ rg warning:
cmd/ci-operator/main.go
534:                    log.Printf("warning: Unable to write JUnit result: %v", err)
723:                            log.Printf("warning: Could not add annotations because you do not have permission to update the namespace (details: %v)", updateErr)
1282:           log.Println("warning: Events will not be created because of lack of permission")

pkg/steps/release/create_release.go
83:                             log.Printf("warning: Unable to lock %s to an image digest pull spec, you don't have permission to access the necessary API.", s.envVar())
90:                     log.Printf("warning: Unable to lock %s to an image digest pull spec due to an import error (%s): %s.", s.envVar(), image.Status.Reason, image.Status.Message)
254:                            log.Printf("warning: Unable to lock %s to an image digest pull spec, you don't have permission to access the necessary API.", s.envVar())
```

Reply:
/shrug